File: BE-FindingJob/src/main/java/com/example/befindingjob/service/BookmarkService.java
package com.example.befindingjob.service;

import com.example.befindingjob.dto.BookmarkRequest;
import com.example.befindingjob.model.ApiResponse;

public interface BookmarkService {
    ApiResponse<Void> bookMark(BookmarkRequest bookmarkRequest);
    ApiResponse<Void> deleteBookmark(BookmarkRequest bookmarkRequest);
}

File: JobHub/app/src/main/java/com/example/jobhub/service/BookmarkService.kt
package com.example.jobhub.service

import com.example.jobhub.dto.BookmarkRequest
import com.example.jobhub.dto.ItemJobDTO
import com.example.jobhub.model.ApiResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST

interface BookmarkService {

    @POST("api/jobseeker/book-mark")
    fun bookMark(@Body bookmarkRequest: BookmarkRequest): Call<ApiResponse<Void>>

    @POST("api/jobseeker/delete-book-mark")
    fun deleteBookMark(@Body bookmarkRequest: BookmarkRequest): Call<ApiResponse<Void>>

    @GET("api/jobseeker/get-all-bookmark-jobs")
    fun getBookmarkedJobsByUser(@Header("token") token: String): Call<ApiResponse<List<ItemJobDTO>>>
}